src/dmd/builtin.d

```diff
@@ -309,6 +309,30 @@ Expression eval_yl2xp1(Loc loc, FuncDeclaration fd, Expressions* arguments)
     return new RealExp(loc, result, arg0.type);
 }
 
+Expression eval_toPrecFloat(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    __gshared float f;
```

Comment:
Why are these `__gshared`?

Reply:
Fear of the optimizer.

Reply:
This comment is I think is relevant to why Martin chose to do it this way.

https://github.com/dlang/druntime/pull/1621#issuecomment-250426515

DMD (and only DMD, at the time it was written) will ignore `cast(float)` if the value is then cast back to a `real`.

Reply:
Please put a note there that says that. Otherwise someone might remove it.

Reply:
I gave non-static intermediates a test, and it would appear my assume is unfounded.

Built both ENABLE_DEBUG and ENABLE_RELEASE builds, and CTFE tests pass just fine.